photon-client/src/main/scala/com/linkedin/photon/ml/io/deprecated/GLMSuite.scala

```diff
@@ -344,13 +344,22 @@ class GLMSuite(
           1
         }
 
+      assert(weight >= 0)
+
       new LabeledPoint(response, features, offset, weight)
     }
 
-    avroRDD.mapPartitions { iter =>
-      val map = _indexMapLoader.indexMapForRDD()
-      iter.map { r => parseAvroRecord(r, map) }
-    }
+    val labeledPoints = avroRDD
+      .mapPartitions { iter =>
+        val map = _indexMapLoader.indexMapForRDD()
+        iter.map { r => parseAvroRecord(r, map) }
+      }
+      .filter(aPoint => aPoint.weight > 0.0)
+
+    // TODO: This is caught later when we check that we have at least 1 training sample
```

Comment:
Add a comment about where it's caught later (Photon Driver.prepareTrainingData; note that the same method is called again from Driver.prepareValidatingData and count is NOT called there)